parsing/parse_hrp.py

```diff
@@ -245,18 +330,29 @@ def parse_hrperf_log(perf_log_path, max_frequency_ghz):
     print("Node memory bandwidth data has been inserted into 'node_memory_bandwidth' table in 'analysis.duckdb'.")
 
 def main():
-    if len(sys.argv) != 3:
-        print('Expected usage: python parse_hrperf_log.py <perf_log_path> <max_frequency_ghz>')
-        sys.exit(1)
+    # Command-line argument parsing
+    parser = argparse.ArgumentParser(description="Parse hiresperf log files and store results in DuckDB.")
+    parser.add_argument('perf_log_path', type=str, help='Path to the hiresperf log file.')
+    parser.add_argument('--raw_counter', action='store_true', help='Includes raw counter data in database.')
+    parser.add_argument('--tsc_ts', action='store_true', help='Use TSC timestamps instead of ktime.')
+    parser.add_argument('--tsc_freq', type=float, required=True, help='TSC frequency in cycles per microsecond.')
 
-    perf_log_path = sys.argv[1]
-    max_frequency_ghz = float(sys.argv[2])
+    args = parser.parse_args()
 
-    if not os.path.isfile(perf_log_path):
-        print(f"Error: File '{perf_log_path}' does not exist.")
+    if not os.path.isfile(args.perf_log_path):
+        print(f"Error: File '{args.perf_log_path}' does not exist.")
         sys.exit(1)
 
-    parse_hrperf_log(perf_log_path, max_frequency_ghz)
+    # Read config flag
+    hrp_use_tsc = read_hrp_tsc_config()
```

Comment:
I do understand the initiative on this check. However, I feel like it might be too strict.

For instance, if we intend to parse a previously collected data, but the header file in the hrperf is already changed. Looks like it will silently proceed without using TSC as the timestamp.

Therefore, instead, I am proposing of just warning the user regarding the mismatch. In the end, it is users' responsibility to validate if collected data is using tsc as the timestamping.